install.sh

```diff
@@ -1,90 +1,349 @@
+show_loader() {
+    local message=$1
+    local pid=$!
+    local delay=0.3
+    local spinstr='|/-\\'
+    while [ "$(ps a | awk '{print $1}' | grep $pid)" ]; do
+        local temp=${spinstr#?}
+        printf "\r%s [%c]  " "$message" "$spinstr"
+        local spinstr=$temp${spinstr%"$temp"}
+        sleep $delay
+    done
+    printf "\r%s [Done]   \n" "$message"
 }
 
-echo "##########################################\nInstalling dependencies\n##########################################"
-# Install dependencies
-if ! command -v node &> /dev/null
-then
-    echo "node could not be found. Please install Node.js 18 or above."
+# Check for Node.js
+echo "Checking for Node.js..."
+if ! command -v node &> /dev/null; then
+    echo "Node.js could not be found. Please install node js 18 or above."
     exit 1
 fi
+
+# Verify Node.js version
 version=$(node -v | cut -d'.' -f1 | tr -d 'v')
-if [ "$version" -lt 18 ]
-then
-  echo "Invalid Node.js version. Expected 18 or above, but got $version."
-  exit 1
+if [ "$version" -lt 18 ]; then
+    echo "Invalid Node.js version. Expected 18 or above, but got $version."
+    exit 1
 fi
+echo "Node.js version is valid."
+
+# Check and Install AWS CLI
+echo "Checking for AWS CLI..."
 if aws --version; then
-  echo "##########################################\nAWS CLI Installed\n##########################################"
+    echo "AWS CLI is already installed."
 else
-  curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
-  sudo installer -pkg AWSCLIV2.pkg -target /
+    echo "Installing AWS CLI..."
+    curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+    sudo installer -pkg AWSCLIV2.pkg -target / & show_loader "Installing AWS CLI.."
+    echo "AWS CLI is installed."
 fi
-if ! command -v aws &> /dev/null
-then
-    echo "AWS CLI could not be found. Please rerun \`sh install.sh\` with Sudo access"
+
+# Check if AWS CLI installation was successful
+if ! command -v aws &> /dev/null; then
+    echo "AWS CLI could not be found. Please rerun 'sh install.sh' with Sudo access."
     exit 1
 fi
-npm install -g aws-cdk
-if ! command -v cdk &> /dev/null
-then
-    echo "AWS CDK could not be found. Please rerun \`sh install.sh\` with Sudo access"
+
+# Install AWS CDK
+echo "Installing AWS CDK..."
+npm install -g aws-cdk & show_loader "Installing AWS CDK..."
+echo "AWS CDK is installed successfully."
+
+# Check for AWS CDK
+if ! command -v cdk &> /dev/null; then
+    echo "AWS CDK could not be found. Please rerun 'sh install.sh' with Sudo access."
     exit 1
 fi
-cdk --version
+
+# Determine OS and run respective dependency script
 os=$(uname)
-if [ "$os" = "Linux" ]; then
-  sh linux_deps.sh
-elif [ "$os" = "Darwin" ]; then
-  sh mac_deps.sh
-else
-  echo "Unsupported operating system."
-  exit 1
-fi
+case "$os" in
+  "Linux")
+    echo "Detecting operating system: Linux"
+    (sh linux_deps.sh & show_loader "Running Linux dependencies script...")
```

Comment:
You might also have to detect the Linux distribution running, since the `linux_deps.sh` script calls `apt-get`, which would be only available on Debian/Ubuntu based systems.

We'll have to either handle more package managers in the `linux_deps.sh` script, or error out from the current one if the user is not running Debian/Ubuntu.

You can refer to this StackExchange question for information on detecting distribution: https://unix.stackexchange.com/q/88644
(Running `cat /etc/*-release` might be the most relevant option nowadays.)

Reply:
Let's update the `linux_deps.sh` script to handle the package managers for installing `jq`on various Linux distributions. Also, let's incorporate support for different machine architectures in the installation process for `kubectl` and `awscli`. For now, we will support a few common architectures and prompt users for manual installation if any issues arise.